src/lazy/encoder/binary/v1_1/container_writers.rs

```diff
@@ -123,110 +275,121 @@ pub struct BinaryStructWriter_1_1<'value, 'top> {
 
 impl<'value, 'top> BinaryStructWriter_1_1<'value, 'top> {
     pub(crate) fn new_length_prefixed(
-        container_writer: BinaryContainerWriter_1_1<'value, 'top>,
+        allocator: &'top BumpAllocator,
+        buffer: &'value mut BumpVec<'top, u8>,
     ) -> Self {
+        const LENGTH_PREFIXED_STRUCT_TYPE_CODE: u8 = 0xC0;
+        const LENGTH_PREFIXED_FLEX_LEN_STRUCT_TYPE_CODE: u8 = 0xFC;
+        let container_writer = BinaryContainerWriter_1_1::new_length_prefixed(
+            LENGTH_PREFIXED_STRUCT_TYPE_CODE,
+            LENGTH_PREFIXED_FLEX_LEN_STRUCT_TYPE_CODE,
+            allocator,
+            buffer,
+        );
         Self {
             flex_uint_encoding: true,
             container_writer,
         }
     }
 
-    pub(crate) fn new_delimited(container_writer: BinaryContainerWriter_1_1<'value, 'top>) -> Self {
+    pub(crate) fn new_delimited(
+        allocator: &'top BumpAllocator,
+        buffer: &'value mut BumpVec<'top, u8>,
+    ) -> Self {
+        const DELIMITED_STRUCT_OPCODE: u8 = 0xF3;
+        let container_writer =
+            BinaryContainerWriter_1_1::new_delimited(DELIMITED_STRUCT_OPCODE, allocator, buffer);
         Self {
             // Delimited structs always use FlexSym encoding.
             flex_uint_encoding: false,
             container_writer,
         }
     }
 
-    pub(crate) fn buffer(&mut self) -> &'_ mut BumpVec<'top, u8> {
-        self.container_writer.buffer
+    pub(crate) fn fields_buffer(&mut self) -> &'_ mut BumpVec<'top, u8> {
+        self.container_writer.child_values_buffer()
     }
+}
 
-    #[inline]
-    pub fn write<A: AsRawSymbolTokenRef, V: WriteAsIon>(
-        &mut self,
-        name: A,
-        value: V,
-    ) -> IonResult<&mut Self> {
-        use RawSymbolTokenRef::*;
+impl<'value, 'top> FieldEncoder for BinaryStructWriter_1_1<'value, 'top> {
+    fn encode_field_name(&mut self, name: impl AsRawSymbolTokenRef) -> IonResult<()> {
+        use crate::RawSymbolTokenRef::*;
```

Comment:
🗺️ `StructWriter` impls now must also define the (private) method `encode_field_name`; they can then defer value serialization to `make_value_writer()`.